mygrad/operation_base.py

```diff
@@ -101,13 +115,30 @@ def backward(self, grad, *, graph, _reduction=None, **kwargs):
         for index, var in enumerate(self.variables):
             if not var.constant:
                 if not var._ops:
-                    raise Exception(
-                        "Invalid Backprop: part of the computational graph containing "
-                        "this tensor was cleared prior to backprop"
+                    raise InvalidBackprop(
+                        "Part of the computational graph containing "
+                        "this tensor was 'cleared' prior to backprop."
+                    )
+                backed_grad = self.backward_var(grad, index, **kwargs)
+
+                if not isinstance(backed_grad, (np.ndarray, Real)) or not np.issubdtype(
+                    np.asarray(backed_grad).dtype, np.number
+                ):
+                    raise InvalidGradient(
+                        "An invalid gradient-value was produced by "
+                        "`{name}.backward_var(..., index={index})`."
+                        "\nGradients are expected to be real-valued scalars or "
+                        "numpy arrays, got a gradient of type: {_type}"
+                        "\nIt is likely that there is an error in the implementation "
+                        "of `{name}.backward_var`.".format(
```

Comment:
when are you dropping 3.5 support? :grimacing:

Reply:
Whenever 3.5 is no longer formally supported :/ (when 3.8 comes out, I think)